### PR TITLE
Add unit tests for DiscoveryHandlers

### LINKED ISSUE #179: Add unit tests for DiscoveryHandlers
## Summary

The `DiscoveryHandlers` class is the **largest handler** in the gateway with 16 methods but has no dedicated unit test file. While extensively covered by integration tests in `test_integration.test.py`, it lacks isolated unit tests for error paths and edge cases.

**Methods to test (16):**

*Areas (5):*
- `handle_list_areas` - list all areas
- `handle_get_area` - get a specific area by ID
- `handle_area_components` - list components within an area
- `handle_get_subareas` - list sub-areas of an area
- `handle_get_contains` - list all entities contained in an area

*Components (5):*
- `handle_list_components` - list all components
- `handle_get_component` - get a specific component by ID
- `handle_get_subcomponents` - list sub-components
- `handle_get_hosts` - list hosting relationships
- `handle_component_depends_on` - list component dependencies

*Apps (3):*
- `handle_list_apps` - list all apps
- `handle_get_app` - get a specific app by ID
- `handle_app_depends_on` - list app dependencies

*Functions (3):*
- `handle_list_functions` - list all functions
- `handle_get_function` - get a specific function by ID
- `handle_function_hosts` - list function hosting

---

## Proposed solution (optional)

Create `test/test_discovery_handlers.cpp`. Due to the size, consider starting with error-path tests that don't require a full entity cache:

```cpp
// Test: invalid entity ID format returns 400
TEST_F(DiscoveryHandlersTest, GetAreaInvalidIdReturns400) { /* ... */ }

// Test: non-existent area returns 404
TEST_F(DiscoveryHandlersTest, GetAreaNotFoundReturns404) { /* ... */ }

// Test: list_areas with empty cache returns empty items array
TEST_F(DiscoveryHandlersTest, ListAreasEmptyCacheReturnsEmptyList) { /* ... */ }

// Test: area_components for non-existent area returns 404
TEST_F(DiscoveryHandlersTest, AreaComponentsNotFoundReturns404) { /* ... */ }
```

This is a larger test file - consider splitting into multiple PRs (areas first, then components, then apps/functions).

### PR DESCRIPTION
**Pull request by eclipse0922**
# Pull Request

## Summary

Add unit coverage for all `DiscoveryHandlers` endpoints using a real `GatewayNode`/`DiscoveryManager` fixture.
Harden the new test fixture with a suite-level node, a dedicated test port, fail-fast temp manifest handling, and explicit `400`/`404` coverage for relationship endpoints.

---

## Issue

Link the related issue (required):

- closes #179

---

## Type

- [ ] Bug fix
- [x] New feature or tests
- [ ] Breaking change
- [ ] Documentation only

---

## Testing

How was this tested / how should reviewers verify it?

- `SKIP=ament-copyright /tmp/pre-commit-host-venv/bin/pre-commit run --files src/ros2_medkit_gateway/CMakeLists.txt src/ros2_medkit_gateway/test/test_discovery_handlers.cpp`
- `docker exec ros2_medkit_issue179 bash -lc 'source /opt/ros/jazzy/setup.bash && cd /home/sewon/workspace && colcon build'`
- `docker exec ros2_medkit_issue179 bash -lc 'source /opt/ros/jazzy/setup.bash && cd /home/sewon/workspace && colcon test && colcon test-result --verbose'`
- `colcon test` currently reports unrelated local failures in `ros2_medkit_fault_reporter` (`test_integration`) and `ros2_medkit_gateway` (`test_gateway_node`, `test_graph_provider_plugin`)
- `test_discovery_handlers` passes locally with 40 tests

---

## Checklist

- [x] Breaking changes are clearly described (not applicable)
- [x] Tests were added or updated if needed
- [x] Docs were updated if behavior or public API changed (not applicable)

### CODE REVIEW

File: src/ros2_medkit_gateway/test/test_discovery_handlers.cpp [hunk 1]
```diff
@@ -0,0 +1,681 @@
+// Copyright 2026 bburda
```
Comment (bburda) [inline src/ros2_medkit_gateway/test/test_discovery_handlers.cpp]: @eclipse0922 You are the author here 🙂
